flybywiresim/a32nx
# feat: flyPad Local Files via SimBridge

## Summary of Changes

Add the following SimBridge dependent features and fixes to the flyPad:

- [x] Local Files
  - [x] PDF
- [x] Fix pagination for PDF
- [x] Fix horizontal and vertical zoom for mixed PDFs (landscape and portrait pages)

Fixes to do:

- [x] Having a subfolder within the resources folders seems to crash simbridge.
  - Requires a fix in SimBridge to be merged (hides folders)] 
- [x] Charts above 5MB in size tend to take very long to display, a loading animation may be needed so that the user knows that the chart is still in the process of loading in.

KnownIssue: 
- Power cycling EFB results in "There is no chart to display" message and inability to display any charts. Workaround: Select Navigraph charts tab, then return to local files tab.

## Screenshots (if necessary)
![image](https://user-images.githubusercontent.com/16833201/185758868-f29d2dce-ed11-452e-b252-1964da4383c4.png)

## Additional context
Requires SimBridge - therefore targeted against the Experimental branch for now.


Discord username (if different from GitHub): Cdr_Maverick#6475

## Testing instructions
Have SimBridge running to test this. 
Add some files to the SimBridge resources/images and resources/pdf folder
Add multi-paged PDF files
- Check that the files you copied into the resource folders appear under Local Files (image, pdf and both)
- Click on files to see of they are loading
- Check if multi-page PDFs are loading and pages can be iterated through
- Check fullscreen, zooming, fit to height, fit to width, rotation, and resetting in any combination you can think of
  - Check this also for Navigraph charts
- Check pinning local files (remember when clicking on a pinned chart and wanting to see all files again, delete the search field content)
- In general try to break anything 


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
